src/AppBundle/Notification/SendOnArticlePublished.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace AppBundle\Notification;
+
+use AppBundle\Entity\Article;
+use Doctrine\Common\EventSubscriber;
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Doctrine\ORM\Event\PreUpdateEventArgs;
+use Doctrine\ORM\Events;
+use GuzzleHttp\Exception\RequestException;
+use Psr\Log\LoggerInterface;
+use sngrl\PhpFirebaseCloudMessaging\ClientInterface;
+use sngrl\PhpFirebaseCloudMessaging\Message;
+use sngrl\PhpFirebaseCloudMessaging\Notification;
+use sngrl\PhpFirebaseCloudMessaging\Recipient\Topic;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+class SendOnArticlePublished implements EventSubscriber
+{
+    const TOPIC = 'news';
```

Comment:
You may already know this, but since the project requires PHP 7.1, you can use `private const TOPIC = 'news';` to make the constant private in case it's only used in this class.